kinto_fxa/authentication.py

```diff
@@ -85,29 +93,39 @@ def _verify_token(self, token, request):
         # `request.bound_data` is an attribute provided by Kinto to store
         # some data that is shared among sub-requests (e.g. default bucket
         # or batch requests)
-        key = 'fxa_verified_token'
-        if key in request.bound_data:
-            return request.bound_data[key]
-
-        # Use PyFxa defaults if not specified
-        server_url = fxa_conf(request, 'oauth_uri')
-        scope = aslist(fxa_conf(request, 'required_scope'))
-        auth_cache = self._get_cache(request)
-        auth_client = OAuthClient(server_url=server_url, cache=auth_cache)
-        try:
-            profile = auth_client.verify_token(token=token, scope=scope)
-            user_id = profile['user']
-        except fxa_errors.OutOfProtocolError as e:
-            logger.exception("Protocol error")
-            raise httpexceptions.HTTPServiceUnavailable()
-        except (fxa_errors.InProtocolError, fxa_errors.TrustError) as e:
-            logger.debug("Invalid FxA token: %s" % e)
+        if REIFY_KEY not in request.bound_data:
+            # Use PyFxa defaults if not specified
+            server_url = fxa_conf(request, 'oauth_uri')
+            auth_cache = self._get_cache(request)
+            auth_client = OAuthClient(server_url=server_url, cache=auth_cache)
+
             user_id = None
+            client_name = None
+
+            for scope, client in request.registry._fxa_oauth_scope_routing.items():
+                try:
+                    profile = auth_client.verify_token(token=token, scope=aslist(scope))
+                    user_id = profile['user']
+                    scope = profile['scope']
+                    client_name = client
+
+                    # Make sure the bearer token scope don't match multiple configs.
+                    if len([x for x in request.registry._fxa_oauth_scope_routing.keys()
```

Comment:
Maybe we could declare `routing_scopes` outside the loop.
Plus we could iterate on it above instead of `request.registry._fxa_oauth_scope_routing.items()`